test/unittest/tests/block/test_block_other.cpp

```diff
@@ -0,0 +1,125 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_other]: block.h
+ * This file unit tests the __wt_block_header_byteswap_copy and __wt_block_eligible_for_sweep
+ * functions.
+ */
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+
+static void
+test_block_header_byteswap_copy(WT_BLOCK_HEADER *from, WT_BLOCK_HEADER *to)
+{
+    WT_BLOCK_HEADER prev_from, prev_to;
+
+    // Save the original values before any potential byte re-orderings.
+    prev_from.disk_size = from->disk_size;
+    prev_from.checksum = from->checksum;
+    prev_to.disk_size = to->disk_size;
+    prev_to.checksum = to->checksum;
+
+#ifdef WORDS_BIGENDIAN
+    __wt_block_header_byteswap_copy(from, to);
+    REQUIRE(to->checksum == __wt_bswap32(prev_from.checksum));
+    REQUIRE(to->disk_size == __wt_bswap32(prev_from.disk_size));
```

Comment:
I think it would be good to also verify that if `from != to`, `from` is not changed by `__wt_block_header_byteswap_copy()`.